packages/patternfly-4/react-core/src/components/Nav/NavToggle.js

```diff
@@ -0,0 +1,41 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+const propTypes = {
+  children: PropTypes.func.isRequired,
+  defaultValue: PropTypes.bool,
+  groupId: PropTypes.number,
+  onToggle: PropTypes.func
+};
+
+const defaultProps = {
+  defaultValue: true,
+  groupId: 0,
+  onToggle: () => undefined
+};
+
+class NavToggle extends React.Component {
+  static propTypes = propTypes;
+  static defaultProps = defaultProps;
+  state = {
+    value: this.props.defaultValue
+  };
+
+  handleToggle = e => {
+    this.setState(({ value }) => ({
```

Comment:
Optional: No need to use the function gimmick of setState, because you do not call it more than once.
```JS
const { value } = this.state;
const { groupId, onToggle } = this.props;
this.setState({
  value: !value;
});
onToggle && onToggle(e, groupId, !value);
```
The function call in setState might confuse someone, with this destructing of state and props you are mutating the state at the very end of function executin. But that's just my preference.